dartagnan/src/main/java/com/dat3m/dartagnan/program/analysis/ExecutionAnalysis.java

```diff
@@ -61,8 +61,14 @@ public boolean isImplied(Event start, Event implied) {
             // If strongest implication does hold, then all progress models will give this implication
             return true;
         }
-        // weakest implication holds but not strongest: progress model decides
-        final boolean implication = switch (progressModel) {
+
+        if (!progressModel.isUniform()) {
+            // For mixed-hierarchy models, we only rely on strongest implication.
+            return strongestImplication; // FALSE
+        }
+
+        // weakest implication holds but not strongest & model is uniform: progress model decides
+        final boolean implication = switch (progressModel.getProgressAtScope("ALL")) {
```

Comment:
Where is this "ALL" coming from? Is it supposed to be `Tag.OpenCL.ALL`? If so, this will not work with other models

Reply:
No, this is in fact arbitrary. Since the hierarchy is uniform, all scopes have the same progress model. I just need to provide any one since I don't have a `getUniformProgress` method or similar.
Though I could see adding a `getDefaultProgressModel` method and using that one here.

Reply:
I added the `getDefaultProgressModel` method.